rayon-core/src/scope/mod.rs

```diff
@@ -511,12 +584,20 @@ impl<'scope> ScopeFifo<'scope> {
 
 impl<'scope> ScopeBase<'scope> {
     /// Creates the base of a new scope for the given worker thread
-    fn new(owner_thread: &WorkerThread) -> Self {
+    fn new(owner: &WorkerThread, registry: Arc<Registry>) -> Self {
         ScopeBase {
-            owner_thread_index: owner_thread.index(),
-            registry: owner_thread.registry().clone(),
+            registry: registry,
             panic: AtomicPtr::new(ptr::null_mut()),
-            job_completed_latch: CountLatch::new(),
+            job_completed_latch: ScopeLatch::new(owner),
+            marker: PhantomData,
+        }
+    }
+
+    fn new_in_place(registry: Arc<Registry>) -> Self {
```

Comment:
I think this would be better named `new_blocking`, since in-place sometimes uses the stealing `new` too.

Reply:
Done